bookmarks/templates/bookmarks/tag_cloud.html

```diff
@@ -20,4 +20,6 @@
             {% endfor %}
         </p>
     {% endfor %}
+    <hr />
+    <a href="?q=!untagged">Untagged</a>
```

Comment:
This should modify the existing `q` param rather than overwriting it, like all the other tag links do. That means if the user has searched for `python` and then clicks on `Untagged`, the search should be `python !untagged` instead of `!untagged`. You can check the existing templates on how this is done. 

As stated in the other comment, it might make sense to remove existing tags from the query as well to prevent mixing `!untagged` and `#sometag` in queries.